Makefile

```diff
@@ -31,6 +31,44 @@ CONTAINER_TOOL ?= docker
 SHELL = /usr/bin/env bash -o pipefail
 .SHELLFLAGS = -ec
 
+# Define the content of the temporary top-most kustomize overlay for the
```

Comment:
@nb-ohad I think we can make is too simple just with one command

```
cd config/manager && kustomize edit add patch --name controller-manager --kind Deployment --patch '[{"op": "replace", "path": "/spec/template/spec/containers/0/env/1/value", "value": "prefix"}]'
```

Also please add the `CSI_SERVICE_ACCOUNT_PREFIX` env variable and its empty default value in the manager.

Pls look at https://github.com/iamniting/ceph-csi-operator/commit/63c09a52b154c9acc4bb518134bab931b5707306

Reply:
@iamniting 
Thank you for the suggestion, but I think you missed the point. We should no edit existing Kustomize files while trying to build. These files are tracked by the repo, which means that any change to them is persistent. 
The more safe way to influence the customization is to build an overlay file and introduce your changes there.

Reply:
I really don't agree with you here that we should not change the existing Kustomize files. Kustomize files should be changed whenever required. We are just introducing an extra layer on top, which I believe is not really necessary.

Reply:
@iamniting Kustomize files should not be personalized locally and then pushed. 
You already did that, I assume by mistake, by running the make target and then committing the changes. That is what created the container image patch inside the manager folder.  

The point is that we should like to have the kustomizes files static and not influenced by local commands running on local branches/forks

Reply:
As this is not about the correctness of the solution but about the approach (which is somewhat a matter of opinion),  I am resolving this so we could merge, to unblock us. We can continue the discussion after merging and if needed we can always pivot with a new PR on top